src/GShark/Geometry/Mesh.cs

```diff
@@ -548,6 +548,138 @@ private FaceData CountFaceEdges()
         }
 
 
+        
+
+        //Function that checks if point is in or out a triangle (boolean result, In - True , Out - False)
+        private bool IsPointInTriangle(Point3 projection, List<Point3> trianglePoints)
```

Comment:
Let's make all list arrays when possible! This will speed up a lot and less resources will be allocated
```suggestion
        private bool IsPointInTriangle(Point3 projection, Point3[] trianglePoints)
```